frontend/packages/core/src/AppProvider/index.tsx

```diff
@@ -129,77 +132,103 @@ const ClutchApp: React.FC<ClutchAppProps> = ({
     [discoverableWorkflows, triggeredHeaderData]
   );
 
+  React.useEffect(() => {
+    if (children) {
+      React.Children.forEach(children, child => {
+        if (React.isValidElement(child)) {
+          const { type } = child?.props || {};
+
+          if (type === "HEADER") {
```

Comment:
Do we have a way of hinting the possibility to use this type prop?, maybe adding a type